sklearn/mixture/_base.py

```diff
@@ -217,6 +217,7 @@ def fit_predict(self, X, y=None):
             Component labels.
         """
         X = _check_X(X, self.n_components, ensure_min_samples=2)
+        self._check_n_features(X, reset=True)
```

Comment:
Shall instead call `_validate_data(X) and move the validation of some parameters (the one in _check_X) right after this validation?

Reply:
or call `validate_data` inside `_check_X`?

Reply:
that would be pre-addressing future PRs (which would call _validate_data in predict, transform, etc)

The scope of the PR is to only call _validate_data in fit, and I tried to keep the changes minimal